File: Atcoder_#352/a_atcoder_line.rb
n, x, y, z = gets.split.map(&:to_i)

# 全駅を配列に変換
stations = (1..n).to_a

# x と y の順序を正規化
x, y = [x, y].min, [x, y].max

# 降りる駅を抽出
getting_off_station = stations[x-1..y-1]

# zが含まれているかを確認
if getting_off_station.include?(z)
  puts "Yes"
else
  puts "No"
end

File: Atcoder_#352/a_atcoder_line.js
const readline = require('readline');

function readInput(callback) {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

  let inputLines = [];

  rl.on('line', (line) => {
    inputLines.push(line);
  }).on('close', () => {
    callback(inputLines);
  });
}

function processInput(input) {
  const [n, x, y, z] = input[0].split(' ').map(Number);

  // 全駅を配列に変換
  let stations = [];
  for (let i = 1; i <= n; i++) {
    stations.push(i);
}


  // x と y の順序を正規化
  const [start, end] = [Math.min(x, y), Math.max(x, y)];

  // 降りる駅を抽出
  const getting_off_station = stations.slice(start - 1, end);

  if (getting_off_station.includes(z)) {
    console.log('Yes');
  } else {
    console.log('No');
  }
}

// 入力を読み取り、コールバック関数で処理する
readInput(processInput);
